Make historical weather more abstract

Rea888 opened the pull request.

Rea888 committed:
Created 2023_02_13_154324_create_historical_humidity_models_table.php renamed from 2023_02_13_154324_create_historical_weather_humidity_models_table.php
Created 2023_05_16_144140_create_historical_temperature_models.php
Created 2023_05_16_145012_create_historical_temperature_processing_reports_models.php
Created AbstractInitializeCommand.php
Created AbstractMeteoData.php
Created AbstractMeteoDataMapper.php
Created AbstractProcessingService.php
Created ApiWeatherClient.php
Added  App\Providers\ElasticLogProvider::class to app.php
Binded GoogleApiClient and MeteoApiClient to AppServiceProvider.php
Created BeginQueueAllModelsCommand.php
Changed php version from 8.2 to 8.1
Added "elasticsearch/elasticsearch": "^8.7", to composer.json
Created CoordinatesDataMapper.php
Created DataIdentifier.php
Created elastic_log.php
Created ElasticLogProvider.php
Created ElasticLogSetup.php
Created FinishQueueAllModelsCommand.php
Created GoogleApiClient.php
Renamed HistoricalWeatherHumidityModel.php to HistoricalHumidityModel.php
Deleted HistoricalHumidityProcessCommand.php
Implemented interface to HistoricalHumidityProcessingReportsModel.php
Deleted HistoricalHumidityProcessingService.php
Created HistoricalTemperatureModel.php
Created HistoricalTemperatureProcessingReportsModel.php
Deleted HistoricalWeatherHumidityCommand.php
Deleted HistoricalWeatherHumidityService.php
Added parameter to HumidityBeginQueueCommand.php
Created HumidityData.php
Created HumidityDataMapper.php
Added parameter to HumidityFinishQueueCommand.php
Created HumidityInitializeCommand.php
Created HumidityProcessingService.php
Created InitializableProcessingServiceInterface.php
Created InitializeAllModelsCommand.php
Fixed command load in commands() in Kernel.php Deleted schedule()
Changed channels from single to daily in logging.php added elastic channel
Deleted LongitudeLatitudeService.php
Deleted LongitudeLatitudeRepository.php
Created MeteoApiClient.php
Created Processable.php interface
Created ProcessedHumidityService.php
Created ProcessedTemperatureService.php
Added string parameter to constructor in ProcessHumidityJob.php
Created ProcessTemperatureJob.php
Added base url to google service in services.php and weather_api service
Created TemperatureBeginQueueCommand.php
Created TemperatureData.php
Created TemperatureDataMapper.php
Created TemperatureFinishQueueCommand.php
Created TemperatureInitializeCommand.php
Created TemperatureProcessingService.php
Added command description to WeatherForecastCommand.php
Changed LongitudeLatitudeRepository.php to GoogleApiClient.php in WeatherForecastController.php
Deleted Route in web.php

Rea888 committed:
Deleted   "arcanedev/log-viewer": "@dev", from composer.json
Deleted  "arcanedev/log-viewer" from composer.lock

Rea888 committed:
Changed for loop to foreach lopp in ProcessedTemperatureService.php nad ProcessedHumidityService.php

KuveeTheBird commented on app/ApiClient/Meteo/MeteoApiClient.php:
```diff
@@ -0,0 +1,84 @@
+        CoordinatesData $coordinatesData
+    ): HumidityData
+    {
+        $preparedQueryParams = $this->prepareQueryParams($startDate, $endDate, $coordinatesData);
```
make the formatting same as in getTemperatureData

KuveeTheBird commented on app/Console/Commands/AbstractInitializeCommand.php:
```diff
@@ -0,0 +1,31 @@
+use DateTime;
+use Illuminate\Console\Command;
+
+class AbstractInitializeCommand extends Command
```
```suggestion
abstract class AbstractInitializeCommand extends Command
```
if classname contains abstract is should be an actual abstract class

KuveeTheBird commented on app/Console/Commands/AbstractInitializeCommand.php:
```diff
@@ -0,0 +1,31 @@
+class AbstractInitializeCommand extends Command
+{
+
+    protected $signature = 'abstract:initialize';
```
```suggestion
```

KuveeTheBird replied:
remove, this is misleading and wrong and unnecessary

KuveeTheBird commented on app/Console/Commands/AbstractInitializeCommand.php:
```diff
@@ -0,0 +1,31 @@
+    {
+        $citiesAndDates = [];
+
+        $startDate = new DateTime(config('city_date.start_date'));
```
use DateTimeImmutable instead of DateTime everywhere

KuveeTheBird commented on app/Console/Commands/AbstractInitializeCommand.php:
```diff
@@ -0,0 +1,31 @@
+            foreach ($cities as $city) {
+                $citiesAndDates[] = new DataIdentifier($city, clone $startDate);
+            }
+            $startDate = $startDate->add(new DateInterval('P1M'));
```
minor optimisation:
new DateInterval('P1M')
can be extracted to a variable outside the while loop

KuveeTheBird commented on app/Console/Commands/BeginQueueAllModelsCommand.php:
```diff
@@ -0,0 +1,22 @@
+
+class BeginQueueAllModelsCommand extends Command
+{
+    protected $signature = 'models:beginQue';
```
```suggestion
    protected $signature = 'models:beginQueue';
```
this feels like you want to "beginWhat" because Que is italian/spanish for what. the word you're looking for is queue :)

KuveeTheBird commented on app/Console/Commands/FinishQueueAllModelsCommand.php:
```diff
@@ -0,0 +1,20 @@
+
+class FinishQueueAllModelsCommand extends Command
+{
+    protected $signature = 'models:finishQue';
```
```suggestion
    protected $signature = 'models:finishQueue';
```

KuveeTheBird commented on app/Console/Commands/Forecast/WeatherForecastCommand.php:
```diff
@@ -1,14 +1,15 @@
 use App\Http\Controllers\WeatherForecastController;
 use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Log;
+use Psr\Log\LoggerAwareInterface;
```
```suggestion
```

you should enable "Optimize imports" in the PHPStorm commit dialog

KuveeTheBird commented on app/Console/Commands/Humidity/HumidityBeginQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @var string
+     */
+    protected $signature = 'humidity:beginQue {parameter}';
```
```suggestion
    protected $signature = 'humidity:beginQueue {parameter}';
```

KuveeTheBird commented on app/Console/Commands/Humidity/HumidityBeginQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+
+    public function handle()
+    {
+        $parameter = $this->argument('parameter');
```
what is this parameter, and what is this used for??

KuveeTheBird commented on app/Console/Commands/Humidity/HumidityFinishQueueCommand.php:
```diff
@@ -0,0 +1,45 @@
+
+    public function handle()
+    {
+        $parameter = $this->argument('parameter');
```
what is this parameter, and what is this used for??

KuveeTheBird commented on app/Console/Commands/Humidity/HumidityFinishQueueCommand.php:
```diff
@@ -0,0 +1,45 @@
+     *
+     * @var string
+     */
+    protected $signature = 'humidity:finishQue {parameter}';
```
```suggestion
    protected $signature = 'humidity:finishQueue {parameter}';
```

KuveeTheBird commented on app/Console/Commands/Temperature/TemperatureBeginQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+
+    public function handle()
+    {
+        $parameter = $this->argument('parameter');
```
what is this parameter, and what is this used for??

KuveeTheBird commented on app/Console/Commands/Temperature/TemperatureBeginQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @var string
+     */
+    protected $signature = 'temperature:beginQue {parameter}';
```
```suggestion
    protected $signature = 'temperature:beginQueue {parameter}';
```

KuveeTheBird commented on app/Console/Commands/Temperature/TemperatureFinishQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+
+    public function handle()
+    {
+        $parameter = $this->argument('parameter');
```
what is this parameter, and what is this used for??

KuveeTheBird commented on app/Console/Commands/Temperature/TemperatureFinishQueueCommand.php:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @var string
+     */
+    protected $signature = 'temperature:finishQue {parameter}';
```
```suggestion
    protected $signature = 'temperature:finishQueue {parameter}';
```

KuveeTheBird commented on app/Service/ApiWeatherClient.php:
```diff
@@ -0,0 +1,27 @@
+
+use App\ApiClient\Google\GoogleApiClient;
+
+class ApiWeatherClient
```
what is this class? do you still need this? if not remove

Rea888 committed:
Changed DateTime to DateTimeImmutable in AbstractInitializeCommand.php
Deleted ApiWeatherClient.php
Deleted parameter in BeginQueueAllModelsCommand.php, FinishQueueAllModelsCommand.php, HumidityBeginQueueCommand.php, HumidityFinishQueueCommand.php, TemperatureBeginQueueCommand.php, TemperatureFinishQueueCommand.php, ProcessedHumidityService.php, ProcessedTemperatureService.php, ProcessHumidityJob.php, ProcessTemperatureJob.php
Changed return type to DateTimeImmutable in getDateTime() function

KuveeTheBird commented:
please double check that all resolved comments are actually done and pushed (hint: they aren't :) )

Rea888 committed:
Fixed typo

Rea888 committed:
Created AbstractProcessingReportsModel.php
Fixed typo in BeginQueueAllModelsCommand.php, FinishQueueAllModelsCommand.php, HumidityBeginQueueCommand.php, HumidityFinishQueueCommand.php, TemperatureBeginQueueCommand.php and TemperatureFinishQueueCommand.php
Deleted unnecessary functions in HistoricalHumidityProcessingReportsModel.php and HistoricalTemperatureProcessingReportsModel.php
Deleted Processable.php interface
Replaced rows with starProcess and finishProcess functions in ProcessedTemperatureService.php and ProcessedHumidityService.php
Deleted unnecessary imports in WeatherForecastCommand.php

Rea888 committed:
Deleted HistoricalWeatherTenYearDatesCreator.php

Rea888 committed:
Deleted HistoricalWeatherTenYearDatesCreator.php
